doc/architecture-decisions/0018-route-crd-and-contour-controller.md

```diff
@@ -0,0 +1,48 @@
+# 18. Add Contour as an Alternative Ingress Solution Provider
+
+Date: 2020-10-09
+
+## Status
+
+Proposal
+
+## Context
+In our efforts to allow more optionality in ingress solutions for CF-for-K8s, we
+want to allow the use of Contour as a potential alternative to Istio.
+
+### Proposed Design
+
+We propose:
+1. Adding a configuration option `ingress_solution_provider` to CF-for-K8s, with
+   potential values `istio` or `contour`.
+2. Creating another kubebuilder controller, similar to routecontroller, to
+   handle when `ingress_solution_provider: contour` is configured.
+3. Renaming `Route Controller` to `Istio Controller`
+
+![Proposed ContourControllerDesign](../assets/contour-controller.png)
+
+### Open Questions
+1. What do we do about the config? Does contour config live in cf-k8s-networking
```

Comment:
My inclination is that the config should live in cf-for-k8s, so we don't have similar pipeline incompatibility issues as we iterate on contour features. We should check that the relint team is fine with it.

Reply:
Yes, great point. We will polish this ADR a bit more and share it with the cf-for-k8s engineers and the community.